jsapp/js/account/security/sso/ssoSection.component.tsx

```diff
@@ -43,32 +44,40 @@ const SsoSection = observer(() => {
           </div>
         </div>
       ) : (
-        <div className={styles.bodySection}>Already connected </div>
+        <div className={styles.bodySection}>{t('Already connected')}</div>
       )}
 
       {socialAccounts.length === 0 ? (
-        <div className={styles.optionsSection}>
-          <a
-            href={
-              PATHS.MS_SSO + '?process=connect&next=%2F%23%2Faccount%2Fsecurity'
-            }
-            className={styles.passwordLink}
-          >
-            <Button
-              label='Set up'
-              size='m'
-              color='blue'
-              type='frame'
-              onClick={() => {
-                /*TODO: Handle NavLink and Button*/
-              }}
-            />
-          </a>
+        <div className={classNames(styles.optionsSection, styles.ssoSetup)}>
+          {socialApps.map((socialApp) => (
+            <a
+              href={
+                'accounts/' +
+                socialApp.provider +
+                '/login/?process=connect&next=%2F%23%2Faccount%2Fsecurity'
+              }
+              className={styles.passwordLink}
+            >
+              <Button
+                label={
+                  socialApps.length == 1
+                    ? t('Set up')
+                    : t('Set up') + ' ' + socialApp.name // translation example: 'Set up Google Apps'
```

Comment:
Let's check if it would be ok to just always show the name. In all places like this.

Reply:
Received the go-ahead, the change is made [here](https://github.com/kobotoolbox/kpi/pull/4508/commits/48f25adab558c040ecf2e635ee8c307b3e974657).